rtf

```diff
@@ -132,33 +136,34 @@ Commands
 """
 
 from __future__ import print_function
+from subprocess import Popen, PIPE
 import json
 import os
-import tempfile
 import shutil
 import copy
 import re
-import scenario.lib
+import sys
 import importlib
-import docopt
-import pexpect
 import signal
 import socket
+import pexpect
 import requests
-from subprocess import Popen,PIPE
-from pprint import pprint
+import docopt
+import scenario.lib
+from vagrant import *
 # Hack to import rudder lib, remove, some day ...
-import sys
-sys.path.insert(0,"./rudder-api-client/lib.python")
+sys.path.insert(0, "./rudder-api-client/lib.python")
 from rudder import RudderEndPoint, RudderError
 
-
 # Run a command in a shell like a script would do
 # And inform the user of its execution
-def shell(command, fail_exit=True, keep_output=True):
+def shell(command, fail_exit=True, keep_output=True, live_output=False):
   if keep_output:
-    process = Popen(command, stdout=PIPE, shell=True, universal_newlines=True)
-    output,error = process.communicate()
+    if live_output:
+      process = Popen(command, shell=True, universal_newlines=True)
+    else:
+      process = Popen(command, stdout=PIPE, shell=True, universal_newlines=True)
+    output, error = process.communicate()
```

Comment:
i think this won't work when you have no pipe at all

Reply:
It is working, doc says it returns None if no stdout are scpecified, which is fine here since we want the output to be print to the shell and not captured by python.https://docs.python.org/2/library/subprocess.html#subprocess.Popen.communicate